### PR TITLE
Support anyOf schemas in the scaffolder template

### PR DESCRIPTION
**Pull request by dhenneke**
This change supports adding custom properties to properties inside of `anyOf` (https://react-jsonschema-form.readthedocs.io/en/latest/usage/oneof/). This can be useful if one wants to provide different sets of defaults that a user can select from. Or different forms depending on whether certain features are enabled or not. In the UI schema, the properties inside of `anyOf` are just flattened.

Example: We have a template that can be targeted to different build environments with a set of known defaults that the user wants to select. This could be done with the following schema:
<details>
<summary>template.yaml</summary>

```yaml
apiVersion: backstage.io/v1beta2
kind: Template
metadata:
  name: a-template
spec:
  type: service
  parameters:
# ...
    - title: Select an Environment Configuration
      anyOf:
      - title: alpha
        properties:
          imageRegistry:
            type: string
            default: my-docker-registry.local
            ui:readonly: true
          enableFossa:
            type: boolean
            default: true
            ui:readonly: true
      - title: omega
        properties:
          imageRegistry:
            type: string
            default: another-docker-registry.local # <- different
            ui:readonly: true
          enableFossa:
            type: boolean
            default: false # <- different
            ui:readonly: true
# ...
  steps:
    - id: fetch-base
      name: Fetch Base
      action: fetch:cookiecutter
      input:
        url: "…"
        values:
          imageRegistry: '{{ parameters.imageRegistry }}'
          enableFossa: '{{ parameters.enableFossa }}'
```
</details>

But there might also be other use-cases. This only works for `anyOf` and not for the other `allOf` or `oneOf` but I haven't got useful use-cases for the other two so I leave it out for now.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/scaffolder/src/components/MultistepJsonForm/schema.ts [hunk 1]
```diff
@@ -39,22 +39,29 @@ function extractUiSchema(schema: JsonObject, uiSchema: JsonObject) {
...
+      const innerUiSchema = {};
+      uiSchema[propName] = innerUiSchema;
+      extractUiSchema(schemaNode, innerUiSchema);
     }
+  }
 
-    const schemaNode = properties[propName];
-    if (!isObject(schemaNode)) {
-      continue;
+  if (Array.isArray(anyOf)) {
```
Comment (Rugvip) [inline plugins/scaffolder/src/components/MultistepJsonForm/schema.ts:58]: Looks like this totally works for `oneOf` and `allOf` as well, wanna add those while at it for completeness?

**Review by Rugvip — APPROVED** [review]
Looks good, thanks! 👍

File: plugins/scaffolder/src/components/MultistepJsonForm/schema.ts (hunk 1 above)
Comment (dhenneke) [reply plugins/scaffolder/src/components/MultistepJsonForm/schema.ts:58]: 👍  I'll add it

**Comment by Rugvip** [comment]
:shipit:

**Pull request merged**